### PR TITLE
Change type checking rules of Fusion ufuncs

### PR DESCRIPTION
**Pull request by asi1024**

### CODE REVIEW

**Comment by asi1024** [comment]
jenkins, test this please.

**Comment by asi1024** [comment]
jenkins, test this please.

File: cupy/core/fusion.py [hunk 1]
```diff
@@ -838,10 +828,12 @@ def wrapper(f, kernel_name=None):
 class ufunc(core.ufunc):
     # Args:
     #   fusion_op (cupy.ufunc)
-    #   cupy_op (cupyu.ufunc or function)
-    #   numpy_op (numpy.ufunc or function)
+    #   cupy_op (cupyu.ufunc or function or None)
```
Comment (toslunar) [inline cupy/core/fusion.py]: Fix typo: `cupyu.ufunc` -> `cupy.ufunc`

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
Please fix test.
```
E     File "/work/cupy/tests/cupy_tests/core_tests/test_fusion.py", line 1535, in test_compile_with_dtypes
E       f.compile_with_dtypes(x.dtype, y.dtype)
E   AttributeError: 'Fusion' object has no attribute 'compile_with_dtypes'
```

**Comment by asi1024** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
LGTM.

**Pull request merged**